backend/app/exporters/models/marc21.rb

```diff
@@ -541,4 +512,70 @@ def handle_ead_loc(ead_loc)
                                 )
   end
 
+  private
+
+    # name fields looks something this:
+    # [["a", "Dick, Philp K."], ["b", nil], ["c", "see"], ["d", "10-1-1980"], ["g", nil], ["q", nil], ["4", "aut"]]
+    def handle_agent_person_punctuation(name_fields)
+      name_fields.sort! {|a, b| a[0][0] <=> b[0][0]}
+
+      #If subfield $c is present, the value of the preceding subfield must end in a comma. 
+      #If subfield $d is present, the value of the preceding subfield must end in a comma. 
+      #If subfield $e is present, the value of the preceding subfield must end in a comma. 
+      ['c', 'd', 'e'].each do |subfield|
+        s_index = name_fields.find_index{|a| a[0] == subfield}
+
+        # check if $subfield is present
+
+        unless !s_index || s_index == 0
+          preceding_index = s_index - 1
+
+          # find preceding field and append a comma if there isn't one there already
+          unless name_fields[preceding_index][1][-1] == "," 
+            name_fields[preceding_index][1] << ","
+          end
+        end
+      end
+
+      #The value of subfield q must be enclosed in parentheses. 
+      q_index = name_fields.find_index{|a| a[0] == "q"}
+      unless !q_index
+        name_fields[q_index][1] = "(#{name_fields[q_index][1]})"
+      end
+
+      #The value of the final subfield must end in a period."
+      unless name_fields[-1][1][-1] == "." 
+        name_fields[-1][1] << "."
+      end
+
+      return name_fields
+    end
+  
+    def gather_agent_person_subfield_mappings(name, role_info)
+      joint = name['name_order'] == 'direct' ? ' ' : ', '
+      name_parts = [name['primary_name'], name['rest_of_name']].reject{|i| i.nil? || i.empty?}.join(joint)
+
+      subfield_e = role_info[0] == "e" ? role_info : nil
+      subfield_4 = role_info[0] == "4" ? role_info : nil
+  
+
+      name_fields = [
```

Comment:
@avatar382 Is the order right here? From the spreadsheet, for person agents I think it is supposed to be this:
For personal names (applies to 100, 600, 700), output in this order: $a primary part of name, rest of name $b number $q (fuller form), $c title, $d dates, $e relator term - so q would be before $c